### PR TITLE
Fix Date, Time issue in Text function when format is not defined

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue:  https://github.com/microsoft/Power-Fx/issues/1054. 
Fix Date, Time issue in Text function when format is not defined.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs [hunk 1]
```diff
@@ -236,33 +236,49 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
...
+                    else if (formatString != null)
                     {
-                        dateTimeValue = FormulaValue.New(d.GetConvertedValue(timeZoneInfo));
+                        return ExpandDateTimeExcelFormatSpecifiers(irContext, formatString, "g", dateValue.GetConvertedValue(timeZoneInfo), culture, runner.CancellationToken);
                     }
                     else
                     {
-                        dateTimeValue = args[0] as DateTimeValue;
+                        resultString = dateValue.GetConvertedValue(timeZoneInfo).ToShortDateString();
                     }
```
Comment (jas-valgotar) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:265]: Can we instead reuse else if part and specify defaultFormat to be "d" ?
[https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-7.0](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-7.0)

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs [hunk 2]
```diff
@@ -236,33 +236,49 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
...
+                    else if (formatString != null)
+                    {
+                        return ExpandDateTimeExcelFormatSpecifiers(irContext, formatString, "g", dtValue.GetConvertedValue(timeZoneInfo), culture, runner.CancellationToken);
+                    }
                     else
                     {
-                        return ExpandDateTimeExcelFormatSpecifiers(irContext, formatString, "g", dateTimeValue.GetConvertedValue(timeZoneInfo), culture, runner.CancellationToken);
+                        resultString = dtValue.GetConvertedValue(timeZoneInfo).ToShortTimeString();
                     }
```
Comment (jas-valgotar) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:282]: same as above:
Can we instead reuse else if part and specify defaultFormat to be "t" ?
[https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-7.0](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-7.0)

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt [hunk 3]
```diff
@@ -213,3 +213,16 @@ Error({Kind:ErrorKind.Div0})
...
+"2/5/2023 12:00 AM"
+
+>> Text(Date(2023,2,5))
+"2/5/2023"
+
+>> Text(Time(2,4,5))
+"2:04 AM"
+
+>> Text(Time(17,2,3))
+"5:02 PM"
```
Comment (jas-valgotar) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt:228]: Can you add below cases please,
`Text(DateValue("01/01/2023"))`
`Text(DateTimeValue("01/01/2023"))`

Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt]: Added more 3 tests: 
>> Text(DateValue("02/05/2023"))
"2/5/2023"

>> Text(DateTimeValue("02/05/2023"))
"2/5/2023 12:00 AM"

>> Text(TimeValue("15:06"))
"3:06 PM"

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs (hunk 2 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:274]: Updated the code. Thx

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs (hunk 1 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:261]: Updated. thx

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by jas-valgotar — APPROVED** [review]

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**